### DISCUSSION TITLE
Reinterpolation hybrid-sigma model level data to a regular altitude grid

### DISCUSSION THREAD

**Discussion by jdassink** [pydata/xarray#7471] [Q&A]
I have an Xarray dataset with the contents of an atmospheric model in hybrid-sigma coordinates. The dataset consists of several model variables that are available on a regular latitude-longitude grid, in this case specified on 65 hybrid-sigma model levels. 

So, each model variable has as dimensions the hybrid-sigma model level number and latitude and longitude.  Consider, for example the temperature field:

```
xarray.DataArray't'hybrid: 65 latitude: 300 longitude: 300
array([[[212.73866, 212.7379 , ..., 212.46098, 212.45708],
        [212.72636, 212.726  , ..., 212.43901, 212.43666],
        ...,
        [279.85785, 279.86017, ..., 278.35107, 278.31335],
        [279.87195, 279.8614 , ..., 278.3589 , 278.1678 ]]], dtype=float32)
```

For each latitude-longitude grid point, I have computed the altitude from the Earth's surface for each hybrid model level number (following [ECMWF's wiki](https://confluence.ecmwf.int/display/CKB/ERA5%3A+compute+pressure+and+geopotential+on+model+levels%2C+geopotential+height+and+geometric+height)) and have stored that as variable 'z'.  

```
xarray.DataArray'z'hybrid: 65 latitude: 300longitude: 300
array([[[3.06163351e+04, 3.06162455e+04, 3.06155455e+04, ...,
         3.05957611e+04, 3.05970100e+04, 3.05964055e+04],
        [3.06158901e+04, 3.06157539e+04, 3.06155182e+04, ...,
         3.05953653e+04, 3.05960723e+04, 3.05961209e+04],
        [3.06157857e+04, 3.06156265e+04, 3.06153596e+04, ...,
         3.05948358e+04, 3.05949636e+04, 3.05955104e+04],
        ...,
        [1.18821703e+01, 1.23908401e+01, 1.19352956e+01, ...,
         1.08624521e+01, 1.23089970e+01, 1.18565123e+01],
        [1.22416687e+01, 1.22158196e+01, 1.20726453e+01, ...,
         1.25545416e+01, 1.26793490e+01, 1.15266169e+01],
        [1.21218788e+01, 1.21214290e+01, 1.23145995e+01, ...,
         1.10582631e+01, 1.15941613e+01, 1.22456452e+01]]])
```

As the hybrid coordinate system is terrain-following, this 3D 'z' variable therefore represents an irregular grid. The irregular altitude grid changes with each lat, lon position. Now, I want to re-interpolate the dataset variables to a regular altitude-grid, making use of the 'z' variable that contains the irregular altitudes. I suppose it involves the `interp()` function, but I cannot see how I can link the model levels (that the hybrid dimension point too) to the values stored in variable 'z'.

Looking forward to your suggestions.

Best,
Jelle

**Comment by dcherian** [comment]
See https://xgcm.readthedocs.io/en/latest/transform.html
